### PR TITLE
🌱 Add tests for VM Condition version conversion

### PR DESCRIPTION
**Pull request by bryanv**
There will likely be some upcoming changes on how the v1a1 PrereqReady condition is handled and filtered out but add some tests first.

```release-note
NONE
```

### CODE REVIEW

File: api/v1alpha1/virtualmachine_conversion.go [hunk 1]
```diff
@@ -668,7 +668,6 @@ func Convert_v1alpha1_VirtualMachineStatus_To_v1alpha2_VirtualMachineStatus(
 
 	out.PowerState = convert_v1alpha1_VirtualMachinePowerState_To_v1alpha2_VirtualMachinePowerState(in.PowerState)
 	out.Network = convert_v1alpha1_Network_To_v1alpha2_NetworkStatus(in.VmIp, in.NetworkInterfaces)
-	out.LastRestartTime = in.LastRestartTime
```
Comment (bryanv) [inline api/v1alpha1/virtualmachine_conversion.go:671]: The autogenerated conversion already handles this.

File: api/v1alpha1/virtualmachine_conversion_test.go [hunk 2]
```diff
@@ -321,6 +322,196 @@ func TestVirtualMachineConversion(t *testing.T) {
...
+			Type:               nextver.VirtualMachineConditionImageReady,
+			Status:             metav1.ConditionTrue,
+			LastTransitionTime: metav1.NewTime(now.AddDate(2, 0, 0)),
+		}
+		vmSetRPCond := metav1.Condition{
+			Type:               nextver.VirtualMachineConditionVMSetResourcePolicyReady,
+			Status:             metav1.ConditionTrue,
+			LastTransitionTime: metav1.NewTime(now.AddDate(3, 0, 0)),
+		}
+		vmBSCond := metav1.Condition{
```
Comment (srm09) [inline api/v1alpha1/virtualmachine_conversion_test.go:346]: `vmBSCond` at a first glance seems to be a **BS** condition like `Foo`, but un fact is **BootstrapReady**. 🤣

**Review by srm09 — APPROVED** [review]
/lgtm

File: api/v1alpha1/virtualmachine_conversion_test.go (hunk 2 above)
Comment (bryanv) [reply api/v1alpha1/virtualmachine_conversion_test.go:356]: It goes along nicely with the `bsArgs` that's used elsewhere. Just a reflection of how I was feeling at the time :)

**Pull request merged**